--- a/include/linux/completion.h
+++ b/include/linux/completion.h
@@ -119,4 +119,13 @@
 extern void complete_on_current_cpu(struct completion *x);
 extern void complete_all(struct completion *);
 
+/**
+ * INIT_COMPLETION - reinitialize a completion structure
+ * @x:  completion structure to be reinitialized
+ *
+ * This macro should be used to reinitialize a completion structure so it can
+ * be reused. This is especially important after complete_all() is used.
+ */
+#define INIT_COMPLETION(x)	((x).done = 0)
+
 #endif
